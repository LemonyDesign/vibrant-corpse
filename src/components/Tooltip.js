import React from 'react';
import '../styles/components/tooltip.scss';
import PropTypes from 'prop-types';
import { Info } from '@material-ui/icons';

function Tooltip({ modifier, baseword }) {
  const { modWord, modDefinition } = modifier;
  const { baseWord, baseDefinition } = baseword;

  return (
    <a className="tooltip" href="#">
      <Info />
      <span className="tooltip-content">
        <span className="tooltip-text">
          <span className="tooltip-inner">
            <span className="wd">{modWord}</span>
            {modDefinition}
            <br />
            <span className="wd">{baseWord}</span>
            {baseDefinition}
          </span>
        </span>
      </span>
    </a>
  );
}

Tooltip.propTypes = {
  modifier: PropTypes.objectOf(PropTypes.string).isRequired,
  baseword: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default Tooltip;
