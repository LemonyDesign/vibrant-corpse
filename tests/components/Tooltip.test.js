import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from '../../src/components/Tooltip';

describe('Tooltip', () => {
  test('should render correctly', () => {
    const mockModifier = {
      modWord: 'green',
      modDefinition: 'this is an adjective',
    };
    const mockBase = {
      baseWord: 'dog',
      baseDefinition: 'this is a noun',
    };

    const tree = renderer.create(<Tooltip modifier={mockModifier} baseword={mockBase} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
