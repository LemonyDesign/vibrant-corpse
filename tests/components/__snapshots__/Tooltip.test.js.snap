// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render correctly 1`] = `
<a
  className="tooltip"
  href="#"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-1"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
  <span
    className="tooltip-content"
  >
    <span
      className="tooltip-text"
    >
      <span
        className="tooltip-inner"
      >
        <span
          className="wd"
        >
          green
        </span>
        this is an adjective
        <br />
        <span
          className="wd"
        >
          dog
        </span>
        this is a noun
      </span>
    </span>
  </span>
</a>
`;
